kotest-assertions/kotest-assertions-yaml/src/commonMain/kotlin/io/kotest/assertions/yaml/YamlMatchers.kt

```diff
@@ -58,8 +60,25 @@ fun equalYaml(
             { "Expected value to be not equal to YAML '$expected', but was: null" }
          )
       } else {
-         val (expectedTree, actualTree) = parse(expected, actual)
-         equalYamlNode(expectedTree).test(actualTree)
+         val actualYaml = try {
+            yaml.parseToYamlNode(actual)
+         } catch (ex: Exception) {
+            return@Matcher MatcherResult(
+               false,
+               { "expected: actual YAML to be valid YAML: $actual" },
```

Comment:
might `ex.message` or the type of Exception contain any useful information to include in the message?

Reply:
Good point. I added the exceptions's message in case of an invalid YAML. I hope it matches the usual text format @AlexCue987